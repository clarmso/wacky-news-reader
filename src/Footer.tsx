import React from "react";

import { Grid } from "@material-ui/core";

import ie from "./img/ms-icon.gif";
import nytimesLogoLight from "./img/poweredby_nytimes_200a.png";
import nytimesLogoDark from "./img/poweredby_nytimes_200c.png";

type FooterProps = {
  themeChoice: string;
  is90s: boolean;
};

const Footer: React.FC<FooterProps> = ({ themeChoice, is90s }) => {
  return (
    <Grid container spacing={5} justify="center">
      <Grid item>
        {themeChoice === "dark" ? (
          <img src={nytimesLogoDark} alt="Powered by New York Times" />
        ) : (
          <img src={nytimesLogoLight} alt="Powered by New York Times" />
        )}
      </Grid>
      {is90s && (
        <Grid item>
          <img src={ie} alt="Best viewed using Internet Explorer" />
        </Grid>
      )}
    </Grid>
  );
};

export default Footer;
