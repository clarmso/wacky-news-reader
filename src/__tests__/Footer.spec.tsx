import { Provider } from "react-redux";
import { createStore } from "redux";
import { render } from "@testing-library/react";
import Footer from "../Footer";
import wackyReducer, { initialState } from "../connect/reducer";

it("renders with initial state", () => {
  const store = createStore(wackyReducer, initialState);
  const { asFragment } = render(
    <Provider store={store}>
      <Footer />
    </Provider>
  );
  const renderedHtml = asFragment();
  expect(renderedHtml).toMatchSnapshot();
});

it.skip("renders with 90s mode", () => {
  const stateWith90s = { ...initialState, is90s: true };
  const store = createStore(wackyReducer, stateWith90s);
  const { asFragment } = render(
    <Provider store={store}>
      <Footer />
    </Provider>
  );
  const renderedHtml = asFragment();
  expect(renderedHtml).toMatchSnapshot();
});
