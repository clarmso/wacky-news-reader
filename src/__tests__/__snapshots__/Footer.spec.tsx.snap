// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with initial state 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-align-items-xs-baseline MuiGrid-justify-xs-space-around"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="https://developer.nytimes.com"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Powered by New York Times"
          src="poweredby_nytimes_200a.png"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
